kernel/include/idt.h

```diff
@@ -58,4 +58,4 @@ extern void isr31();
 
 void init_idt();
 
-void idt_set_gate(uint8_t num, uint32_t base, uint16_t sel, uint8_t flags);
+void idt_set_gate(int num, uint32_t base, uint16_t sel, uint8_t flags);
```

Comment:
In the `stdint.h` standard file, there's a declaration (in my case at line 48).

`typedef unsigned char           uint8_t;`

So the type `uint8_t` is an `unsigned char`, if we try to compile, the compiler will show this message:

> kernel/cpu/idt.c:71:2: warning: array subscript has type 'char' [-Wchar-subscripts]
>   idt_entries[num].always0 = 0;

In that case i've set the `idt` index, like an int, because it's an `integer`, not a `char`, I've used it and the compilation works.